obspy/io/nied/fnetmt/core.py

```diff
@@ -0,0 +1,258 @@
+        return _buffer_proxy(filename_or_buf, __is_fnetmt_catalog,
+                             reset_fp=True)
+    # Happens for example when passing the data as a string which would be
+    # interpreted as a filename.
+    except (OSError):
+        return False
+
+
+def __is_fnetmt_catalog(buf):
+    """
+    Test whether file is an F-net moment tensor catalog file by reading the
+    header and the first data line. Reads at most 40 lines.
+
+    :param buf: File to read.
+    :type buf: Open file or open file like object.
+    """
+    cnt = 0
+    try:
+        while True:
+            line = buf.readline()
+            if not line:
+                return False
+            line = line.decode()
+            # read at most 40 lines
+            if cnt > 40:
+                return False
+            if line.find('Total Number') != -1:
+                match = re.search(r'Total Number:\s+(\d+)\s+', line)
+                if match:
+                    nevents = int(match.group(1))
+            if line.startswith('Origin Time'):
+                if nevents > 0:
+                    data = buf.readline()
+                    a = data.split()
+                    if len(a) != 21:
+                        return False
+                    return True
+            cnt += 1
+    except:
+        return False
+    else:
+        return True
+
+
+def _read_fnetmt_catalog(filename_or_buf, **kwargs):
+    """
+    Reads an F-net moment tensor catalog file to a
+    :class:`~obspy.core.event.Catalog` object.
+
+    :param filename_or_buf: File to read.
+    :type filename_or_buf: str or file-like object.
+    """
+    return _buffer_proxy(filename_or_buf, __read_fnetmt_catalog, **kwargs)
+
+
+def __read_fnetmt_catalog(buf, **kwargs):
+    """
+    Reads an F-net moment tensor catalog file to a
+    :class:`~obspy.core.event.Catalog` object.
+
+    :param buf: File to read.
+    :type buf: Open file or open file like object.
+    """
+    events = []
+    cur_pos = buf.tell()
+
+    # This also works with BytesIO and what not.
+    buf.seek(0, 2)
+    size = buf.tell()
+    buf.seek(cur_pos, 0)
+
+    # First read the headerlines containing the data request parameters
+    headerlines = []
+    while True:
+        if buf.tell() >= size:
+            break
+        line = buf.readline().decode()
+        if line.find('Total Number') != -1:
+            match = re.search(r'Total Number:\s+(\d+)\s+', line)
+            if match:
+                nevents = int(match.group(1))
+        if line.startswith('Origin Time'):
+            break
+        headerlines.append(line)
+    cur_pos = buf.tell()
+
+    # Now read the catalog
+    while True:
+        if buf.tell() >= size:
+            break
+        line = buf.readline().strip()
+        # If there is something, jump back to the beginning of the line and
+        # read the next event.
+        if line:
+            buf.seek(cur_pos, 0)
+            events.append(__read_single_fnetmt_entry(buf))
```

Comment:
Maybe I'm missing something, but this function is called only from here and only reads one line, so why can't you just pass the line you just read to it?